pkg/virt-handler/vm.go

```diff
@@ -2561,6 +2564,72 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationTarget(origVMI *v1.Vir
 	}
 	return nil
 }
+
+func ConfigureHousekeepingCgroup(vmi *v1.VirtualMachineInstance, d *VirtualMachineController) error {
+	cgroupManager, err := cgroup.NewManagerFromVM(vmi)
+	if err != nil {
+		return err
+	}
+
+	exists, err := cgroupManager.CreateChildCgroup("housekeeping", "cpuset")
+	// bail out if cgroup already configured
+	if exists == true {
+		return nil
+	}
```

Comment:
IIUC, this means that we determine which threads should be attached to the housekeeping cgroup only once when we first identify a VM.

My question is: is it possible that more threads will be created later on?
If so, does it make sense to simply skip the child cgroup creation and setting the CPU set and jump straight to attaching the relevant threads into the housekeeping cgroup?

Reply:
Yes, it it possible that more threads will be created later on, but since they would be children of their parent, they would inherit the parents cgroup (which is the housekeeping cgroup).

Reply:
We'd like any additional threads (for example any threads executed "in the" pod with k exec ..) to also join the housekeeping cgroup, uncertain how that should be fixed.

Reply:
That's an interesting question.
Who's the father of the process that created due to a `k exec`? Or in other words - in which cgroup would it be created?

Reply:
runc i suppose? Unsure.

Anyway it will join the pod cgroup, which has "cpumask" as all pcpus allocated to the pod cgroup.
It should instead be in the housekeeping cgroup.

Reply:
That's interesting.
Maybe @vasiliy-ul has an idea?

Reply:
I don't think there is a straightforward way to move the processes spawned by `k exec ...` to a custom cgroup. Probably they can be moved afterwards. Just as a thought: I remember there was a `freezer` controller... if it was possible to freeze all the exec'ed tasks by default, then it would be possible to move them to housekeeping and unfeeze.

Reply:
> I don't think there is a straightforward way to move the processes spawned by `k exec ...` to a custom cgroup. Probably they can be moved afterwards. Just as a thought: I remember there was a `freezer` controller... if it was possible to freeze all the exec'ed tasks by default, then we could move them to housekeeping and unfeeze. But that might be a crazy idea :P

:)
The problem with moving the new `k exec` processes after they're created is that these calls tend to die pretty quickly. I'm not sure if it would even be effective to move them, unless we're talking about consistent access to the guest's shell.

BTW: is it possible to automatically pin new processes into a pCPU once they spawn into the cgroup? Something like "all processes under this cgroup are bound to this and that pCPU"? Maybe we should do things backwards: make a "dedicated CPUs" sub-cgroup for vCPU pinning and let all other processes run in the main pod cgroup? WDYT? @vladikr

Reply:
> The problem with moving the new k exec processes after they're created is that these calls tend to die pretty quickly. I'm not sure if it would even be effective to move them, unless we're talking about consistent access to the guest's shell.

Then, probably, we need to first define what sort of `k exec ...` processes we want to target and what the use case is. From my side that was just a thought... but I probably fail to see the need for that.

Reply:
"Maybe we should do things backwards: make a "dedicated CPUs" sub-cgroup for vCPU pinning and let all other processes run in the main pod cgroup? WDYT? @vladikr"

That won't work: the main pod cgroups cpumask must be a superset of the "dedicated CPUs" sub-cgroup. For example:

pod cgroup cpumask: 1,2,3,4
dedicated CPUs sub-cgroup: 1,2

So any process that is attached to the pod cgroup will be able to execute on all CPUs of any sub-cgroup.

Reply:
"Then, probably, we need to first define what sort of k exec ... processes we want to target and what the use case is. From my side that was just a thought... but tbh I fail to see the need for that. What do we want to achieve actually?"

All k exec processes. The use-case is: if the user desires to execute "k exec ..." for any reason, then that process or processes should not execute on the isolated pCPUs where the isolated vCPUs execute.